portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,16 +42,17 @@ public Message(String sndr, String msg){
             message = msg;
         }
     }
-    
+    List<Message> messages = new ArrayList<>();
+    int numberOfComments = 0;
```

Comment:
if numberOfComments refer to the number of entries in "messages" array list, you don't need it. You can simply call messages.size() to get that. That way, you only have one single source of truth.